### PR TITLE
refactor: improve coverage

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

router, EggCore

##### Description of change

- fix error when instantiate EggCore
- fix named route when use regexp

### CODE REVIEW

File: lib/utils/router.js [hunk 1]
```diff
@@ -189,52 +190,52 @@ class Router extends KoaRouter {
...
-            replacedParams.push(key);
-            return utility.encodeURIComponent(Array.isArray(values) ? values[0] : values);
-          }
-          return $0;
-        });
-        for (const key in args) {
-          if (replacedParams.indexOf(key) !== -1) {
-            continue;
-          }
+    assert(!is.regExp(url), `Can't get the url for regExp ${url} for by name '${name}'`);
```
Comment (popomore) [inline lib/utils/router.js]: @huacnlee 原来就不支持正则么？rails 怎么做的。

Comment (popomore) [inline lib/utils/router.js]: @atian25 刚才的问题就是这里怎么处理

File: lib/utils/router.js [hunk 2]
```diff
@@ -255,7 +256,7 @@ class Router extends KoaRouter {
 function splitedRouterParams(args) {
   let prefix;
   let middlewares;
-  if (args.length >= 3 && is.string(args[1])) {
+  if (args.length >= 3 && (is.string(args[1]) || is.regExp(args[1]))) {
```
Comment (popomore) [inline lib/utils/router.js]: 原来的 bug @dead-horse

File: lib/utils/router.js (hunk 1 above)
Comment (huacnlee) [inline lib/utils/router.js]: Rails route 的实现部分我还没看过

**Comment by popomore** [comment]
测试修复了

**Pull request merged**